File: src/main/java/kr/mjc/rachel/basics/jdbc/post/ListPosts.java
package kr.mjc.rachel.basics.jdbc.post;

import kr.mjc.rachel.basics.jdbc.DaoFactory;
import lombok.extern.slf4j.Slf4j;

import java.util.List;
import java.util.Scanner;

@Slf4j
public class ListPosts {
  public static void main(String[] args) {
    PostDao postDao = DaoFactory.getPostDao();
    Scanner scanner = new Scanner(System.in);
    System.out.print("count page : ");
    int count = scanner.nextInt();
    int page = scanner.nextInt();
    scanner.close();

    List<Post> postList = postDao.listPosts(count, page);
    log.debug(postList.toString());
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/springjdbc/post/ListPosts.kt
package kr.mjc.jacob.spring.springjdbc.post

import kr.mjc.jacob.jdbc.Page
import kr.mjc.jacob.jdbc.post.PostDao
import kr.mjc.jacob.spring.springjdbc.applicationContext
import java.util.*

fun main() {
  val postDao = applicationContext.getBean(PostDao::class.java)

  print("List posts - pageNumber(0~) pageSize ? ")
  val page = Scanner(System.`in`).use { Page(it.nextInt(), it.nextInt()) }

  val postList = postDao.list(page)
  postList.forEach(::println)
}